File: April-18/17-04-18/java/01-Monk-and-real-estate.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int t;
        t = input.nextInt();
        while(t-- > 0)
        {
            int e = input.nextInt();
            int hash[] = new int[10002];
            int cnt = 0;
            while(e-- > 0)
            {
                int a = input.nextInt();
                int b = input.nextInt();
                if(hash[a] == 0)
                {
                    hash[a] = 1;
                    cnt++;
                }
                if(hash[b] == 0)
                {
                    hash[b] = 1;
                    cnt++;
                }
            }
            System.out.println(cnt);
        }
    }
}

File: April-18/17-04-18/python/01-Monk-and-real-estate.py
for _ in range(int(input())):
    ar = [0]*10002
    cnt = 0
    for __ in range(int(input())):
        a, b = map(int,input().split())
        if(ar[a] == 0):
            ar[a] = 1
            cnt += 1
        if(ar[b] == 0):
            ar[b] = 1
            cnt += 1
    print(cnt)
